Feature/1686 button width

danyball opened the pull request:
Fixes #1686 

# Done is when (DoD):
- [x] Modifications within components are reflected by tests.
- [x] A self-review of the code has been performed.
- [x] The modified component properties have been added to the typescript typings.
- [x] Potential design changes have been reviewed by a designer.
- [x] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [x] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [x] The modifications are shortly added to CHANGELOG.md with the issue number.
- [x] The modifications still work in IE.
- [x] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

danyball committed:
Add possibility to set css width

danyball committed:
Beautify code preview

danyball committed:
Changelog

danyball committed:
Changelog

raphaellueckl commented on src/components/10-atoms/button/demo.js:
```diff
@@ -11,6 +11,7 @@ storiesOf('Components|Atoms/Button/Demos', module)
       sidebar: Readme,
     },
     changelog: Changelog,
+    knobs: { disabled: true },
```
nice!

LucaMele commented on src/components/10-atoms/button/CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## 4.1.0
+
+- You can now set css `width` and the `button` always look quite nice. (#1686).
```
@danyball: You are able to set the css `width` property and the `button` looks always nice.

01231 replied:
Don't like the expression "quite nice". It is not measurable and the user doesn't know what really changed. I would rewrite it.

danyball replied:
Changed it to `nice`. Or do you want  more technical details?

01231 replied:
Or something like: "You can now set a custom css `width` for the `button` to fit your needs"

danyball replied:
You could do that before. But now the button looks good with it. I could also write that now the label is centered.

01231 replied:
Ah okay, then choose what describes it the best in your opinion👍🏼

raphaellueckl reviewed (dismissed):
naise!

danyball committed:
Changelog

01231 commented on src/components/10-atoms/button/demo.js:
```diff
@@ -68,4 +69,14 @@ storiesOf('Components|Atoms/Button/Demos', module)
+  .add(
+    'Button css width',
+    () =>
+      '<axa-button style="width: 80%">Next step</axa-button><axa-text>You can simply set width by setting css width.</axa-text>'
```
Please mention the width that is set (`width: 80%`) so that the user sees what's going on.

danyball committed:
Changed button labels to see technical details

danyball committed:
Take prettierconfig to code-preview-prettier rules

danyball committed:
Set display to main element;
Set height of hover animation to 100% of parents width to see the animation even on very large buttons

danyball committed:
Refactor demo

danyball commented:
New fixes since your last review @NoahLiechti @LucaMele @raphaellueckl : 
- hover animation on large buttons are not over full width
- buttons side by side were shifted
- refactored button demos

01231 reviewed (approved):
Great job!
